src/Planner/PlannerJoinTree.cpp

```diff
@@ -1216,7 +1216,11 @@ JoinTreeQueryPlan buildQueryPlanForTableExpression(QueryTreeNodePtr table_expres
                     const auto & data_header = query_plan.getCurrentHeader();
                     if (!data_header->findByName(static_cast<std::string_view>("_table")))
                     {
-                        const auto & table_name = storage->getStorageID().getTableName();
+                        String table_name;
+                        if (table_node && !table_node->getTemporaryTableName().empty())
```

Comment:
I would prefer surrounding the argument to the ! operator with parentheses for clarity:

```suggestion
                        if (table_node && !(table_node->getTemporaryTableName().empty()))
```